.github/workflows/preview.yml

```diff
@@ -0,0 +1,99 @@
+name: Preview PR
+
+on:
+  workflow_run:
+    workflows: ["Build and publish"]
+    types: 
+      - completed
+
+jobs:
+# This workflow contains a single job called "build"
+  build:
```

Comment:
is there a way to mark the job to only be done on PRs, so that we can avoid checking `        if: ${{ github.event.workflow_run.event == 'pull_request' }}` everywhere?

Reply:
Not when trigger is workflow run (unless the trigger event is strictly PR, which is not in our case). Though an early step that terminates execution of rest if not PR could be used.